src/Platform/Microsoft.Testing.Platform/Hosts/ServerTestHost.cs

```diff
@@ -450,13 +444,20 @@ private async Task<ResponseArgsBase> ExecuteRequestAsync(RequestArgsBase args, s
 
         DateTimeOffset adapterLoadStart = _clock.UtcNow;
 
+        IPlatformOutputDevice outputDevice = ServiceProvider.GetRequiredService<IPlatformOutputDevice>();
+        // TODO: What if some one wants to have their own output device that's also compatible with server mode in IDE?
```

Comment:
I think here we need to do something better. 

Despite `IPlatformOutputDevice` is a bit special and the implementation needs to take into  account more scenario(like app restart and correct handling for the "output") I'd like to abstract it from the "modes". 

`IPlatformOutputDevice` implements `IOutputDevice` and today we pass to the `IOutputDevice` requestor the `IPlatformOutputDevice` implementation as-is. 

Clearly this is not more good if we need to route display info due to modes.

So we should when  `IOutputDevice` is requested from an extension proxy it to be able to forward to the `IPlatformOutputDevice` but also to other internal part of the platform that needs it, for instance the server mode.

High level idea
```
 GetService<IOutputDevice>() -> OutputDeviceRouter -> PlatformOutputDeviceImplementation
                                                                                        -> ServerModeForwarder
```

We cannot forward the `PlatformOutputDeviceImplementation` display..but I would expect that the info printed by the "device" are not important...the device output goal is to "show information" in "some way" it doesn't affect and should not affect in any way the execution, so information produced by this extension shouldn't be so useful.

cc: @Evangelink